shared/src/business/utilities/DateHandler.js

```diff
@@ -135,8 +138,30 @@ const isValidDateString = (dateString, formats = ['M-D-YYYY', 'M/D/YYYY']) => {
   return moment(dateString, formats, true).isValid();
 };
 
+/**
+ * Calculates the difference in calendar days between timeStamp1 and timeStamp2
+ * When timeStamp1 is greater (more recent) than timeStamp2, the difference
+ * will be positive. Time-stamps occurring on the same day will yield zero.
+ * If timeStamp2 is greater than timeStamp1, the result will be negative.
+ *
+ * @param {string} timeStamp1 an ISO-8601 date string
```

Comment:
This is a great question! I'm doing some testing on Joi's date validation: https://hapi.dev/module/joi/api/?v=17.1.1#dateiso

Reply:
Dates are constructed by the system based on separate MM DD YYYY inputs from users or the current time.

These strings pass validation:

- `"2020-01-06T12:00:00.000Z"`
- `"2020-04-01T00:00:00Z"`
- `"2020-04-01T00:00:00+00:00"`

These do not pass validation despite being valid ISO 8601 dates based on the spec:

- `"2020-W14"`
- `"04-01-2020"`
- `"20200401T000000Z"`

These results mach up with what I was expecting. Would you like to see some tests to call this out?

Reply:
Yes, I think it would be useful to have some tests to document what kinds of date strings are allowed and what are not allowed if it isn't fully ISO-8601-compliant. 
I can also see where having some checks on any dates entering the system (APIs especially) to make sure dates conform to the allowed formats and returning errors if not will be critical to enabling data coherency as part of a data migration/import effort as well.